src/js/actions/transform.js

```diff
@@ -564,7 +564,10 @@ define(function (require, exports) {
                 return descriptor.endTransaction(transaction);
             })
             .then(function () {
-                return this.transfer(layerActions.resetIndex, document);
+                var boundsPromise = this.transfer(layerActions.resetBounds, document, document.layers.allSelected),
+                    indexPromise = this.transfer(layerActions.resetIndex, document);
```

Comment:
In this case, both dispatches dispatch by ID, so it should be fine. History also gets amended correctly for both cases, since they don't cause changes in PS history.